File: api/v2/api/client/dto.py
from ninja import Schema
from pydantic import Field, UUID4


class ApiClientDTO(Schema):
    client_id: UUID4 = Field(..., description="Client ID of the API client")
    name: str = Field(..., description="Name of the API client")
    redirect_uri: str | None = Field(None, description="Redirect URI of the API client")

File: frontend/src/Api/dto.ts
export interface ApiClient {
    id: number;
    name: string;
    client_id: string;
    redirect_uri: string | null;
}

export interface UserToken {
    id: number;
    client: ApiClient | null;
    created_at: string | null;
}

export interface CreateUserToken extends UserToken {
    token: string;
}
